src/varname.jl

```diff
@@ -77,26 +93,55 @@ Return the indexing tuple of the Julia variable used to generate `vn`.
 
 ```jldoctest
 julia> getindexing(@varname(x[1][2:3]))
-((1,), (2:3,))
+(@lens _[1][2:3])
 
 julia> getindexing(@varname(y))
 ()
 ```
 """
 getindexing(vn::VarName) = vn.indexing
 
+"""
+    get(obj, vn::VarName{sym})
+
+Alias for `get(obj, PropertyLens{sym}() ∘ vn.indexing)`.
+"""
+function Setfield.get(obj, vn::VarName{sym}) where {sym}
+    return Setfield.get(obj, PropertyLens{sym}() ∘ vn.indexing)
+end
+
+"""
+    set(obj, vn::VarName{sym}, value)
+
+Alias for `set(obj, PropertyLens{sym}() ∘ vn.indexing, value)`.
+"""
+function Setfield.set(obj, vn::VarName{sym}, value) where {sym}
+    return Setfield.set(obj, PropertyLens{sym}() ∘ vn.indexing, value)
+end
+
 
 Base.hash(vn::VarName, h::UInt) = hash((getsym(vn), getindexing(vn)), h)
 Base.:(==)(x::VarName, y::VarName) = getsym(x) == getsym(y) && getindexing(x) == getindexing(y)
 
-function Base.show(io::IO, vn::VarName)
+# Composition rules similar to the standard one for lenses, but we need a special
+# one for the "empty" `VarName{..., Tuple{}}`.
+Base.:∘(vn::VarName{sym,<:IdentityLens}, lens::Lens) where {sym} = VarName{sym}(lens)
+Base.:∘(vn::VarName{sym,<:Lens}, lens::Lens) where {sym} = VarName{sym}(vn.indexing ∘ lens)
```

Comment:
The comment is obsolete now; and [`compose` specializes on `IdentityLens`](https://github.com/jw3126/Setfield.jl/blob/3ce508dfaf680e7aa31997b61568a75913a69ea7/src/lens.jl#L132).  Additionally, shouldn't we rather add a method to `Setfield.compose` directly?

```suggestion
Setfield.compose(vn::VarName{sym,<:Lens}, lens::Lens) where {sym} = VarName{sym}(vn.indexing ∘ lens)
```

Reply:
Sure, this seems like a good idea :+1:

And the `IdentityLens` dispatch was really just there for when we could have either `Lens` or `Tuple`, not to try and "optimize" away anything (I didnt' check, but assumed that it indeed specialized on `IdentityLens`). But this is obsolete now:)

Reply:
Actually we need to overload `\circ` not `Setfield.compose` because `Setfield.compose` will only dispatch if `\circ` is called with lenses as arguments.